lib/ansible/modules/dnf.py

```diff
@@ -347,7 +347,11 @@
 from ansible.module_utils.yumdnf import YumDnf, yumdnf_argument_spec
 
 try:
-    import dnf
+    user_lc_all, os.environ['LC_ALL'] = os.environ['LC_ALL'], 'C'
```

Comment:
use `get_best_parsable_locale` instead

`from ansible.module_utils.common.locale import get_best_parsable_locale`

Reply:
also you might want to wrap `respawn_module` call

Reply:
> use `get_best_parsable_locale` instead
> 
> `from ansible.module_utils.common.locale import get_best_parsable_locale`
> 
> though you might need to set it in `main()` after instantiating the `module`

Using `get_best_parsable_locale` is the plan, it's why I asked https://github.com/ansible/ansible/pull/75250#discussion_r670236365

Reply:
responded there, the python `locale` module does not give the info we need (locales present and compiled, ready for use)